src/components/tooltip/GCMarker.js

```diff
@@ -423,80 +430,103 @@ function _treeInsert(
   let node = tree.get(component);
   if (!node) {
     // Make a new node and grow the tree in this direction.
-    node = { branches: new Map() };
+    const value = { name: '(temp)', time: 0 };
+    node = { branches: new Map(), leaf: false, value };
 
     tree.set(component, node);
   }
 
   if (path.length > 0) {
     // There are more path components.  This node should be a branch if it
     // isn't one already.
-    if (node.value) {
-      // We delete the value to change this node from a leaf to a branch.
-      delete node.value;
-    }
+    node.leaf = false;
     _treeInsert(node.branches, path, phase);
   } else {
     // Make the new node leaf node.
-    if (node.value) {
+    if (node.value.name !== '(temp)') {
       console.error(
-        'Duplicate phases in _treeInsert in MarkerTooltipContents.js'
+        `Duplicate phase ${phase.name} in _treeInsert in MarkerTooltipContents.js`
       );
       return;
     }
     node.value = phase;
+    node.leaf = true;
```

Comment:
I started out doing it that way, but at least at the time, it ended up being more confusing. But at that time, I was still allowing `value` to be optional, so maybe if I went back to it then it would still work? I'm not sure.

Then again, I also recall that neither way felt clearly better, so maybe it's not worth bothering with.